src/taipy/core/data/pickle.py

```diff
@@ -83,11 +86,11 @@ def __init__(
             edit_in_progress,
             **properties,
         )
-        self.__is_file_generated = False
-        self._pickle_path = self.__build_path()
-        if not self._last_edit_date and os.path.exists(self._pickle_path):
+        if self._path is None:
+            self._path = self.__build_path()
```

Comment:
these two lines should be executed before executing this: `properties[self.__PATH_KEY] = self._path`. No?

Reply:
because the __build_path() needs the `self.id` attribute so we must put it after the `super().init()` call. We don't need to save the path anyway because it is based on the `id`.

Reply:
If we save the path, the next time the data node is loaded, the `is_file_generated` will return False, which makes the pickle file not cleaned when we delete the data node. I think it is better to not save to make the behavior consistent

Reply:
you are right. So I would propose to set it in the properties so it is stored and can be retrieved.

```
if properties is None:
    properties = {}
default_value = properties.pop(self.__DEFAULT_DATA_KEY, None)
if not file_generated:= properties.get(self.__FILE_GENERATED_KEY, None):
    properties[self.__FILE_GENERATED_KEY]=False
super().__init__(
    config_id,scope,id,name,parent_id,last_edit_date,job_ids,validity_period,  
    edit_in_progress,**properties
)
 self._path = properties.get(self.__PATH_KEY, properties.get(self.__DEFAULT_PATH_KEY))
 if self._path is None:
     self._path = self.__build_path()
 if self._path is not None:
     properties[self.__PATH_KEY] = self._path
``` 

Then it must be updated in both the `is_file_generated` property and in the `__build_path` method as well.

What do you think?

Reply:
This field `is_generated` is still useful.
The purpose of it is to know if the path has been generated by taipy, or if the file has been created by a user. So when we delete the entity we don't delete the file if it has not been generated.

Reply:
I mean if the path properties is None then we already know it is generated, so saving it is kinda redundant.

Reply:
Yes but what if the path is not known? How do we know if it has been generated by Taipy or if it comes from the user? Either we just use the `is_generated` field, or we need to run a regexp on the path to see if it is generated or not (which is much more complex).